uportal-war/src/main/java/org/jasig/portal/groups/db/ExternalDBGroupStore.java

```diff
@@ -0,0 +1,345 @@
+                log.warn("Waiting for tasks to respond to being cancelled");
+                if (!pool.awaitTermination(5, TimeUnit.SECONDS))
+                    log.error("Pool did not terminate");
+//                    System.err.println("Pool did not terminate");
+              }
+            } catch (InterruptedException ie) {
+              // (Re-)Cancel if current thread also interrupted
+              pool.shutdownNow();
+              // Preserve interrupt status
+              Thread.currentThread().interrupt();
+            }
+            
+//        scheduler.getScheduledExecutor().shutdownNow();
+        scheduler.shutdown();
+        executor.shutdown();
+        try {
+            if (spring_context instanceof AbstractApplicationContext)
+                ((AbstractApplicationContext)this.spring_context).close();
+        } finally {
+            super.finalize();
+        }
+        
+    }       
+
+    
+    private final Log log = LogFactory.getLog(getClass());
+    
+//    public static final String ROOT_KEY = "db_root";
+    public static final String ROOT_KEY = "DB Root";
+    public static final String ROOT_DESC = "A root group provided for the ExternalDBGroupStore.";
+    public static final IEntityGroup ROOT_GROUP = createRootGroup();
+    
+    @Value(value = "${GroupStore.GroupQuery}")
+    private String query = "";
+    @Value(value = "${GroupStore.GroupMemberQuery}")
+    private String groupMemberQuery = "";
+    @Value(value = "${GroupStore.GroupMembershipQuery}")
+    private String groupMembershipQuery = "";
+    @Value(value = "${GroupStore.GroupEntityMembersQuery}")
+    private String groupEntityMembersQuery = "";
+    private ApplicationContext spring_context = null;
+    private final Map<String,IEntityGroup> groups = new HashMap<String, IEntityGroup>();
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+    
+    
+    @Scheduled(cron = "${GroupStore.update.cron}")
+    private void updateSchedule() {
+        this.updateGroups();
+    }
+    
+    @Async
+    protected void updateGroups() {
+        List<Map<String, Object>> groupList = jdbcTemplate.queryForList(query);
+            
+            Map<String, IEntityGroup> newGroups  = new HashMap<String, IEntityGroup>();
+            
+            for (Map<String,Object> groupDef : groupList) {
+                IEntityGroup group = new EntityTestingGroupImpl((String) groupDef.get("group_name"), IPerson.class);
+                group.setName((String) groupDef.get("group_name"));
+                group.setDescription((String) groupDef.get("group_name"));
+                newGroups.put(group.getName(), group);
+            }
+            
+            synchronized (groups) {
+                groups.clear();
+                groups.putAll(newGroups);
+            }
+            log.warn(groups);
+    }
+    
+    @Override
+    public boolean contains(IEntityGroup group, IGroupMember member) throws GroupsException {
```

Comment:
Two questions about this method:

- Can this flavor of groups contain child groups?  (Can those groups be other flavors?)  If not, you may want to check for those qualities and `return false` quickly
- Doesn't this method get called a lot?  Is there a way to implement this method that doesn't involve a database query for every invocation?